Tools/paletteable sprite editor/js/marvinj-0.9.js

```diff
@@ -0,0 +1,2665 @@
+					
+					if(min == -1 || color < min){
+						min = color;
+					}
+					if(max == -1 || color > max){
+						max = color;
+					}
+						
+					average+=color;
+					pixels++;
+				}
+			}
+		}
+		
+		average /= pixels;
+			
+		var color = image.getIntComponent0(x,y);
+		
+		if(color < average*thresholdPercentageOfAverage || (max-min) <= 30){
+			imageOut.setIntColor(x, y, 255, 0, 0, 0);
+		} else{
+			imageOut.setIntColor(x, y, 255, 255, 255, 255);
+		}
+	};
+
+	function CombineByAlpha(){
+		MarvinAbstractImagePlugin.super(this);
+		this.load();
+	}
+	
+	CombineByAlpha.prototype.load = function(){
+		this.setAttribute("x", 0);
+		this.setAttribute("y", 0);
+		this.setAttribute("imageOther", null);
+	}
+	
+	CombineByAlpha.prototype.process = function
+	(
+		imageIn, 
+		imageOut,
+		attributesOut,
+		mask, 
+		previewMode
+	)
+	{
+		var imageOther = this.getAttribute("imageOther");
+		var x = this.getAttribute("x");
+		var y = this.getAttribute("y");
+		
+		if(imageOther != null){
+			
+			for(var j=0; j<imageIn.getHeight(); j++){
+				for(var i=0; i<imageIn.getWidth(); i++){
+					
+					var ox = i-x;
+					var oy = j-y;
+					
+					if(ox >= 0 && ox < imageOther.getWidth() && oy >= 0 && oy < imageOther.getHeight()){
+						var alpha = imageOther.getAlphaComponent(ox,oy);
+						if(alpha != 0){
+							var factor = alpha/255;
+							
+							var rA = imageIn.getIntComponent0(i,j);
+							var gA = imageIn.getIntComponent1(i,j);
+							var bA = imageIn.getIntComponent2(i,j);
+							
+							var rB = imageOther.getIntComponent0(ox,oy);
+							var gB = imageOther.getIntComponent1(ox,oy);
+							var bB = imageOther.getIntComponent2(ox,oy);
+							
+							var red = Math.floor( (rA * (1-factor)) + (rB * (factor)) );
+							var green = Math.floor( (gA * (1-factor)) + (gB * (factor)) );
+							var blue = Math.floor( (bA * (1-factor)) + (bB * (factor)) );
+							
+							imageOut.setIntColor(i,j,Math.max(imageIn.getAlphaComponent(x,y), alpha), red, green, blue);
+						} else{
+							imageOut.setIntColor(i,j, imageIn.getIntColor(i,j));
+						}
+					} else{
+						imageOut.setIntColor(i,j, imageIn.getIntColor(i,j));
+					}
+				}
+			}
+		}
+	};
+	function Convolution(){
+		MarvinAbstractImagePlugin.super(this);
+		this.load();
+	}
+	
+	Convolution.prototype.load = function(){
+		this.setAttribute("matrix", null);
+	}
+	
+	Convolution.prototype.process = function
+	(
+		imageIn, 
+		imageOut,
+		attributesOut,
+		mask, 
+		previewMode
+	)
+	{
+		var matrix = this.getAttribute("matrix");
+		
+		if(matrix != null && matrix.length > 0){
+			
+			for(var y=0; y<imageIn.getHeight(); y++){
+				for(var x=0; x<imageIn.getWidth(); x++){
+					
+					if(y >= matrix.length/2 && y < imageIn.getHeight()-matrix.length/2 && x >= matrix[0].length/2 && x < imageIn.getWidth()-matrix[0].length/2){
+						this.applyMatrix(x, y, matrix, imageIn, imageOut);
+					}
+					else{
+						imageOut.setIntColor(x, y, 0xFF000000);
```

Comment:
Codacy found an issue: [The numeric literal '0xFF000000' will have at different value at runtime.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6395467)